core/sail/nativerdf/src/main/java/org/eclipse/rdf4j/sail/nativerdf/datastore/DataFile.java

```diff
@@ -197,29 +203,50 @@ public byte[] getData(long offset) throws IOException {
 				(data[2] << 8) & 0x0000ff00 |
 				(data[3]) & 0x000000ff;
 
-		// We have either managed to read enough data and can return the required subset of the data, or we have read
-		// too little so we need to execute another read to get the correct data.
-		if (dataLength <= data.length - 4) {
+		// If the data length is larger than 750MB, we are likely reading the wrong data. Probably data corruption.
```

Comment:
how did you come up with 750MB? Is this a special size?

Reply:
It was what I experienced when testing.

I wouldn't think that a single literal would realistically be 750 MB. But I'll double check that the logic only applies if the system property is activated.

Reply:
I added some more context to the comment